test/minikube/minikube_base_database_test.go

```diff
@@ -236,9 +232,14 @@ func restoreDatabase(t *testing.T, namespaceName string, podName string, databas
 	restore := func() {
 		testlib.InjectTestValues(t, options)
 		helm.Install(t, options, testlib.RESTORE_HELM_CHART_PATH, restName)
-		testlib.AddTeardown(testlib.TEARDOWN_RESTORE, func() { helm.Delete(t, options, restName, true) })
+		defer helm.Delete(t, options, restName, true)
```

Comment:
whats the difference between deleting it here and deleting it later?

Reply:
Well, clearly we want the 'helm.delete', and 'kubectl delete' to occur directly one after the other.
This code avoids the dependence on the testlib, which is a good thing.
I agree I should have added a comment at the time as to why I used 'defer' rather than a teardown.

My memory is that I chose what I felt was the simplest solution.
In this case, defer works fine, so there was no need to use a teardown.

Reply:
I do not follow. We always try to minimize the number of changes and the code churn. Every change should be motivated by a real improvement.

So my question is:
- why is the delete required now and why was it not required before? Deleting a chart deletes all k8s objects from that chart.